internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -729,6 +736,19 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context,
 		update = true
 	}
 
+	if clusterInstall.Spec.DiskEncryption != nil {
+		params.DiskEncryption = &models.DiskEncryption{}
+		if cluster.DiskEncryption == nil { // true when current cluster configuration does not include disk encryption
+			cluster.DiskEncryption = &models.DiskEncryption{}
+		}
+		updateString(swag.StringValue(clusterInstall.Spec.DiskEncryption.EnableOn), swag.StringValue(cluster.DiskEncryption.EnableOn), &params.DiskEncryption.EnableOn)
+		updateString(swag.StringValue(clusterInstall.Spec.DiskEncryption.Mode), swag.StringValue(cluster.DiskEncryption.Mode), &params.DiskEncryption.Mode)
+		if clusterInstall.Spec.DiskEncryption.TangServers != cluster.DiskEncryption.TangServers {
```

Comment:
Can't also use updateString for this?

Reply:
hurts my eye as well, but no. `params.DiskEncryption.TangServers` is a string where this func expects `*string` as the third parameter.

Reply:
oh, bummer:( maybe we could add another func later for this, like updateStringValue :)